src/List.js

```diff
@@ -1,119 +1,154 @@
   const [showAlert, setShowAlert] = useState(false);
   const [showingAllTasks, setShowingAllTasks] = useState(true);
-
-  function handleAdd(e) {
-    console.log("adding");
-    setData([...data, {
-      id: idCounter,
-      task: ""
-    }]);
-    setIsEditingId(idCounter);
-    setEditingText("");
-    setIdCounter(idCounter + 1);
+  const [newestItem, setNewestItem] = useState(null);
+
+  useEffect(() => {
+    const allSnap = props.collection.get();
+    allSnap.then(snap => setTotalTasks(snap.size));
+    const checkedSnap = props.collection.where('checked', '==', true).get();
+    checkedSnap.then(snap => setNumChecked(snap.size));
+  }, [props.collection]);
+
+  function handleAdd() {
+    const newId = generateUniqueID();
+    const newTask = {
+      id: newId,
+      created: firebase.database.ServerValue.TIMESTAMP,
+      task: '',
+      priority: '3',
+      checked: false
+    };
+    props.collection.doc(newId).set(newTask);
+    setNewestItem(newId);
   }
 
   function handleEditClick(e) {
-    setIsEditingId(e.target.id);
+    console.log('handleeditclick');
     setEditingText(e.target.value);
+    setIsEditingId(e.target.id);
+    setNewestItem(null);
   }
 
   function handleEditChange(e) {
+    console.log('handleeditchange');
     setEditingText(e.target.value);
-    let newData = data.map(item =>
-      parseInt(item.id) === parseInt(isEditingId)
-      ? { ...item, task: editingText }
-      : item);
-    setData(newData);
+    props.collection.doc(e.target.id).set({ task: editingText }, { merge: true });
   }
 
   function handleEditComplete(e) {
-    let newData = data.map((item) =>
-      parseInt(item.id) === parseInt(isEditingId)
-        ? { ...item, task: editingText }
-        : item
-    );
-    setData(newData);
-    setEditingText("");
+    props.collection.doc(e.target.id).set({ task: editingText }, { merge: true });
     setIsEditingId(null);
   }
 
-  function handleEditEnter (e) {
-    if (e.key === "Enter") {
+  function handleEditEnter(e) {
+    if (e.key === 'Enter') {
+      handleEditComplete({ target: { id: e.target.id } });
       e.target.blur();
-      handleEditComplete(e);
     }
   }
 
-  function handleIsCheckedChange(e) {
-    let newId = parseInt(e.target.id);
-    if (isChecked.includes(newId)) {
-      setIsChecked(isChecked.filter((id) => id !== newId));
-    } else {
-      setIsChecked([...isChecked, newId]);
-    }
+  async function handleIsCheckedChange(e) {
+    const docRef = props.collection.doc(e.target.id);
+    const doc = await docRef.get();
+    const newCheckedState = !doc.data().checked;
+    await docRef.update({ checked: newCheckedState });
   }
 
   function handleToggleAlert() {
     setShowAlert(!showAlert);
   }
 
-  function handleRemoveAllClick() {
-    setData(data.filter(e => !isChecked.includes(e.id)));
-    setIsChecked([]);
+  function handlePriorityChange(e) {
+    props.collection.doc(e.target.id).set({ priority: e.target.value }, { merge: true });
```

Comment:
Why not just use update (like you did in line 69) rather than set with merge=true?